game/squadrons/squadron.py

```diff
@@ -43,6 +43,15 @@ class Squadron:
         default_factory=list, init=False, hash=False, compare=False
     )
 
+    #: A separate list of pilots for the autoplanner.
+    #: The pilots will be reserved from here during each turn when running the autoplanner,
+    #: before actual aircraft orders are made. This will prevent the autoplanner from making
+    #: several overlapping orders for the same squadron, potentially causing the number of
+    #: ordered aircraft from exceeding the number of available pilots.
+    autoplannable_pilots: list[Pilot] = field(
```

Comment:
How does this differ in behavior from `available_pilots`? This sounds like the same thing. The _name_ is definitely describing the same thing. "Available" means "available to the auto planner" (and the player, but since the auto planner runs before the user _can_ interact, that's not a meaningful difference).

This implementation seems way overcomplicated to me. I think all we need is to check (during procurement) that the number of expected aircraft next turn does not exceed the number of expected pilots next turn. What am I missing that makes that not work?

Reply:
The key difference is that if the reservations are made during the autoplanner mission proposal stage, which means the autoplanner will select a different squadron if all pilots are already reserved from the first one. That will not happen if the impose the limit only at the budget spending stage, because substitute squadrons will not be looked for (during this turn, anyway) and the purchases would simply be scrubbed.

My understanding about available_pilots was that it's also used outside the autoplanner stage, which is why I took that as the data source for the pilot reservation list but implemented a separate data structure. It can, of course, be refactored but I would need to study how the available pilot handling works outside the autoplanner stage.

Reply:
> That will not happen if the impose the limit only at the budget spending stage, because substitute squadrons will not be looked for (during this turn, anyway) and the purchases would simply be scrubbed.

That seems like something that should be fixed in the procurer though. When purchases happen pilots are already assigned, so the procurer has all the information it needs to know that a squadron shouldn't be considered and that it should move on to one that does have pilots available.

Reply:
I'll take a look at refactoring this with the actual check for pilots done in the procurer. My hunch is that it will actually be a lot simpler that way. I'll probably get around to do that either tomorror or the day after.